internal/controller/k0smotron.io/k0smotroncluster_statefulset.go

```diff
@@ -200,6 +200,12 @@ func (r *ClusterReconciler) generateStatefulSet(kmc *km.Cluster) (apps.StatefulS
 		if kmc.Spec.Persistence.PersistentVolumeClaim.Name == "" {
 			kmc.Spec.Persistence.PersistentVolumeClaim.Name = kmc.GetVolumeName()
 		}
+
+		if kmc.Spec.Persistence.AutoDeletePVCs {
+			statefulSet.Spec.PersistentVolumeClaimRetentionPolicy = &apps.StatefulSetPersistentVolumeClaimRetentionPolicy{
+				WhenDeleted: apps.RetainPersistentVolumeClaimRetentionPolicyType,
```

Comment:
```suggestion
				WhenDeleted: apps.DeletePersistentVolumeClaimRetentionPolicyType,
```
As I have understood the problem, shouldn't this type of removal policy be followed if the flag is true?

Reply:
Heh, great catch, thanks! Fixed!